relay-metrics/src/protocol.rs

```diff
@@ -164,20 +165,92 @@ fn is_valid_name(name: &str) -> bool {
     false
 }
 
-/// Validates an MRI of the form `<ty>:<ns>/<name>@<unit>`
+/// The namespace of a metric.
 ///
-/// Note that the format used in the statsd protocol is different: Metric names are not prefixed
-/// with `<ty>:` as the type is somewhere else in the protocol.
-pub(crate) fn is_valid_mri(name: &str) -> bool {
-    let mut components = name.splitn(2, ':');
-    if components.next().is_none() {
-        return false;
+/// This successfully deserializes any kind of string, but in reality only `"sessions"` and
+/// `"transactions"` is supported. Everything else is dropped both in the metrics aggregator and in
+/// the store actor.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum MetricNamespace {
+    /// Metrics extracted from sessions.
+    Sessions,
+    /// Metrics extracted from transaction events.
+    Transactions,
+    /// Metrics that relay doesn't know the namespace of, and will drop before aggregating.
+    ///
+    /// We could make this variant contain a string such that customer and PoP-relays can forward
```

Comment:
This is a nice code comment if you prefer to keep it but should better not go into public docs. We have plenty of enums following this pattern though.

Reply:
I shortened this comment to not talk too much about impl